File: app/src/main/java/com/afterlogic/aurora/drive/presentation/modules/fileView/interactor/FileViewImageItemInteractorImpl.java
package com.afterlogic.aurora.drive.presentation.modules.fileView.interactor;

import android.net.Uri;
import com.afterlogic.aurora.drive.core.common.rx.ObservableScheduler;
import com.afterlogic.aurora.drive.data.modules.files.FilesDataModule;
import com.afterlogic.aurora.drive.data.modules.files.repository.FilesRepository;
import com.afterlogic.aurora.drive.model.AuroraFile;
import com.afterlogic.aurora.drive.presentation.common.modules.model.interactor.BaseInteractor;
import io.reactivex.Single;
import java.io.File;
import javax.inject.Inject;
import javax.inject.Named;

/**
 * Created by sashka on 16.02.17.<p/>
 * mail: sunnyday.development@gmail.com
 */

public class FileViewImageItemInteractorImpl extends BaseInteractor implements FileViewImageItemInteractor {

    private final FilesRepository mFilesRepository;
    private final File mCacheDir;

    @Inject FileViewImageItemInteractorImpl(ObservableScheduler scheduler,
                                            FilesRepository filesRepository,
                                            @Named(FilesDataModule.CACHE_DIR) File cacheDir) {
        super(scheduler);
        mFilesRepository = filesRepository;
        mCacheDir = cacheDir;
    }

    @Override
    public Single<Uri> viewFile(AuroraFile file) {
        return mFilesRepository.viewFile(file)
                .compose(this::composeDefault);
    }
}

File: app/src/main/java/com/afterlogic/aurora/drive/presentation/modules/fileView/interactor/FileViewImageItemInteractorImpl.kt
package com.afterlogic.aurora.drive.presentation.modules.fileView.interactor

import android.net.Uri
import com.afterlogic.aurora.drive.core.common.rx.ObservableScheduler
import com.afterlogic.aurora.drive.data.modules.files.repository.FilesRepository
import com.afterlogic.aurora.drive.model.AuroraFile
import com.afterlogic.aurora.drive.presentation.common.modules.model.interactor.BaseInteractor
import io.reactivex.Single
import javax.inject.Inject

/**
 * Created by sashka on 16.02.17.
 *
 *
 * mail: sunnyday.development@gmail.com
 */

class FileViewImageItemInteractorImpl @Inject internal constructor(
        scheduler: ObservableScheduler,
        private val mFilesRepository: FilesRepository
) : BaseInteractor(scheduler), FileViewImageItemInteractor {

    override fun viewFile(file: AuroraFile): Single<Uri> {
        return mFilesRepository.viewFile(file)
                .compose { composeDefault(it) }
    }

}
